packages/ember-runtime/lib/system/core_object.js

```diff
@@ -976,6 +960,50 @@ function injectedPropertyAssertion() {
   assert('Injected properties are invalid', validatePropertyInjections(this));
 }
 
+function flattenProps({ concatenatedProperties, mergedProperties }, props) {
+  let hasConcatenatedProps = concatenatedProperties && concatenatedProperties.length > 0;
```

Comment:
Please tweak this to be something more like:

```js
let hasConcatenatedProps = concatenatedProperties !== undefined && concatenatedProperties.length > 0;
```

(same for `mergedProperties` just below)